code/drasil-code/lib/Language/Drasil/Code/Imperative/Modules.hs

```diff
@@ -381,26 +406,31 @@ constraintViolatedMsg q s c = do
 -- | Generates statements to print descriptions of constraints, using words and
 -- the constrained values. Constrained values are followed by printing the
 -- expression they originated from, using printExpr.
-printConstraint :: (OOProg r) => ConstraintCE ->
+printConstraint :: (OOProg r) => CodeVarChunk -> ConstraintCE ->
```

Comment:
This feels like a bad change. For one, you're forcing a dummy `CodeVarChunk` be created when you're printing a `Range`. Second, you don't want a `CodeVarChunk`, all you want is a name -- why don't you instead ask for a name?

This is also a case where you might want to look upstream at who calls `printConstraint` and see if the design of all of this is correct.

Reply:
I can change it to use a String not a CodeVarChunk since I still need some way to get the set's name.

Reply:
`String` would be a good start. Using `UID` for digging out a `String` is an anti-pattern, BTW. That's what things like `shortname` are for.